### DISCUSSION TITLE
Error ts(2307) while importing a PNG image with "url:..."

### DISCUSSION THREAD

**Discussion by ghost** [parcel-bundler/parcel#5194] [Q&A] [answered]
I am getting a ts error that says:

> Cannot find module 'url:../assets/parcel.png' or its corresponding type declarations. ts(2307)

![ts error while import static image](https://i.stack.imgur.com/2Wc0p.png)

The image is bundled correctly, but I keep getting this ts error.

I have the following tsconfig.json:

```json
{
    "compilerOptions": {
        "jsx": "react",
        "esModuleInterop": true
    },
}
```

Since Parcel bundles without issues, I am assuming this is related to some missing TypeScript configuration. Do you confirm this?

**Comment by mischnic** [comment, marked as answer]
Maybe adding something like this to your project makes TS happy?

```ts
declare module 'url:*' {
  export default string;
}
```

**Reply by ghost** [reply]
It sure did. Thanks. I've put it in a `.d.ts` file in the root folder, but where should it be exactly?

**Reply by mischnic** [reply]
It doesn't really matter, maybe some .d.ts file at the top of your project directory.

https://www.typescriptlang.org/docs/handbook/modules.html#working-with-other-javascript-libraries

**Reply by ghost** [reply]
As a collaborator would you agree with `ambientModules.d.ts`?

**Reply by mischnic** [reply]
I would have probably called it `parcel.d.ts` 🤷  Again, TS doesn't care at all.

**Reply by ghost** [reply]
Thank you!

**Reply by KokoDoko** [reply]
If I follow the above instructions I get the following error:
```
'string' only refers to a type, but is being used as a value here.ts(2693)
```

**Reply by bergwerf** [reply]
In vscode I had to create a `tsconfig.json` with explicit includes, otherwise the linter would only pick up the module declaration as long as the file was opened. I also got the above error `'string' only refers to a type, but is being used as a value here.`, so I only included `declare module 'url:*'`.

**Reply by kim3er** [reply]
` "skipLibCheck": true` in your 'tsconfig.json' should resolve @KokoDoko's error.

**Comment by ghost** [comment]
Hey @mischnic I'd like to build onto this answer, because now I am trying to add type-checking to a project and have followed [your instructions](https://v2.parceljs.org/languages/typescript/#type-checking) but TS keeps complaining about not finding the module 'url:../assets/parcel.png' or its type declaration. NOTE: I have the `parcel.d.ts` file with the content you provided above.

**EDIT:** After having added `"./parcel.d.ts"` to the `"include"` array in `tsconfig.json` TS no longer seems to be complaining. **Still, I wonder if these instructions are really necessary, because type-checking seems to be working just fine without `@parcel/validator-typescript`.** Could you confirm this, please?

**Comment by sigmaSd** [comment]
The docs now says to use
```ts
declare module 'url:*' {
  const value: string;
  export default value;
}
```
